python/examples/ingestion_with_csv/main.py

```diff
@@ -0,0 +1,108 @@
+from sift_py.ingestion.config.telemetry import TelemetryConfig
+from sift_py.ingestion.flow import FlowConfig, FlowOrderedChannelValues
+from sift_py.ingestion.service import IngestionService
+
+
+def _parse_csv(
+    path_to_csv: Path, telemetry_config: TelemetryConfig
+) -> List[FlowOrderedChannelValues]:
+    flows: List[FlowOrderedChannelValues] = []
+
+    flow = telemetry_config.flows[0]  # Packed into a single flow for this example
+    flow_name = flow.name
+
+    with open(path_to_csv, "r") as csv_file:
+        reader = csv.reader(csv_file)
+        next(reader)  # skip header
+
+        for row in reader:
+            timestamp_str, values = row[0], row[1:]
+            assert len(values) == len(
+                flow.channels
+            ), "number of channels don't match number of data points in row"
+
+            flows.append(
+                {
+                    "flow_name": flow_name,
+                    "timestamp": datetime.fromisoformat(timestamp_str),
+                    "channel_values": [double_value(float(raw_value)) for raw_value in values],
+                }
+            )
+
+    return flows
+
+
+def _load_telemetry_config(
+    path_to_csv: Path, asset_name: str, ingestion_client_key: str
+) -> TelemetryConfig:
+    channels = []
+
+    with open(path_to_csv, "r") as csv_file:
+        reader = csv.reader(csv_file)
+        header = next(reader)  # grab only header
+        channel_names = header[1:]
+
+        for name in channel_names:
+            channels.append(
+                ChannelConfig(
+                    name=name,
+                    data_type=ChannelDataType.DOUBLE,  # Assuming all channels are doubles for this example
+                )
+            )
+
+    return TelemetryConfig(
+        asset_name=asset_name,
+        ingestion_client_key=ingestion_client_key,
+        flows=[FlowConfig(name="data", channels=channels)],
+    )
+
+
+if __name__ == "__main__":
+    """
+    Example of ingesting data from a CSV file into Sift.
+    """
+
+    load_dotenv()
+
+    sift_uri = os.getenv("SIFT_API_URI")
+    assert sift_uri, "expected 'SIFT_API_URI' environment variable to be set"
+
+    apikey = os.getenv("SIFT_API_KEY")
+    assert apikey, "expected 'SIFT_API_KEY' environment variable to be set"
+
+    asset_name = os.getenv("ASSET_NAME")
+    assert asset_name, "expected 'ASSET_NAME' environment variable to be set"
+
+    ingestion_client_key = os.getenv("INGESTION_CLIENT_KEY")
+    assert ingestion_client_key, "expected 'INGESTION_CLIENT_KEY' environment variable to be set"
+
+    sample_data_csv = Path("sample_data.csv")
+
+    telemetry_config = _load_telemetry_config(sample_data_csv, asset_name, ingestion_client_key)
+    flows = _parse_csv(sample_data_csv, telemetry_config)
+
+    sift_channel_config = SiftChannelConfig(
+        uri=sift_uri,
+        apikey=apikey,
+    )
+
+    with use_sift_channel(sift_channel_config) as channel:
```

Comment:
Oh actually it looks like we're also missing the actual ingestion step. You'll need to send the flows you created from the `parse_csv` function to `ingest_flows` method (I think that's what it's called). The docs should have that info :)

Reply:
[02e89ec](https://github.com/sift-stack/sift/pull/96/commits/02e89ec3b790cb788dcbdcfa51ecb9c6a40deb64) did I get that right? Since we're already in a context manager with the `with` that will clean things up for us when the ingest is done?

Reply:
You will also have to use buffered ingestion within a `with` block as well, I think?

Reply:
Like so? [3e18495](https://github.com/sift-stack/sift/pull/96/commits/3e18495ea16e1fb840113ca1636600abf40aec4b)

Reply:
Almost! I think you should be able to do `buffered_ingestion.try_ingest_flows(*flows)`. I would double check the function signature.

Reply:
Oh yeah, I completely missed that! Looks like that's the case 👍

Reply:
Thanks, updated! Will go ahead a re-request review